src/Nethermind/Nethermind.Blockchain.Test/BlockTreeTests.cs

```diff
@@ -433,9 +433,13 @@ public void Find_headers_reverse()
             Block block0 = Build.A.Block.WithNumber(0).TestObject;
             Block block1 = Build.A.Block.WithNumber(1).WithParent(block0).TestObject;
             Block block2 = Build.A.Block.WithNumber(2).WithParent(block1).TestObject;
+            Block block3 = Build.A.Block.WithNumber(3).WithParent(block2).TestObject;
+            Block block4 = Build.A.Block.WithNumber(4).WithParent(block3).TestObject;
             AddToMain(blockTree, block0);
             AddToMain(blockTree, block1);
             AddToMain(blockTree, block2);
+            AddToMain(blockTree, block3);
+            AddToMain(blockTree, block4);
```

Comment:
Why this did not catch the issue before? Should you update this test somehow on line 444?

Reply:
BlockTree didn't have any blocks past the requested one, which caused `endBlock` - how it was previously calculated to always be `null`